docs/plugin_developer_guide.md

```diff
@@ -1,96 +1,96 @@
-# Bluemix CLI Plugin Developer's Guide
+# Bluemix CLI Plug-in Developer's Guide
 
```

Comment:
Changed "plugin" to "plug-in" in the text throughout to match IBM Style.

Note that I did NOT change it in actually command snippets, since it's fine to use just "plugin" there.